pkg/pillar/cmd/baseosmgr/baseosmgr.go

```diff
@@ -79,52 +78,39 @@ type baseOsMgrContext struct {
 	rebootImage              string    // Image from which the last reboot happened
 }
 
-var debug = false
-var debugOverride bool // From command line arg
+func newBaseOsMgrContext() *baseOsMgrContext {
+	ctx := baseOsMgrContext{}
+	ctx.globalConfig = types.DefaultConfigItemValueMap()
 
-func Run(ps *pubsub.PubSub) {
-	versionPtr := flag.Bool("v", false, "Version")
-	debugPtr := flag.Bool("d", false, "Debug flag")
-	flag.Parse()
-	debug = *debugPtr
-	debugOverride = debug
-	if debugOverride {
-		log.SetLevel(log.DebugLevel)
-	} else {
-		log.SetLevel(log.InfoLevel)
-	}
-	if *versionPtr {
-		fmt.Printf("%s: %s\n", os.Args[0], Version)
-		return
-	}
-	agentlog.Init(agentName)
-	if err := pidfile.CheckAndCreatePidfile(agentName); err != nil {
-		log.Fatal(err)
-	}
+	ctx.agentBaseContext = agentbase.DefaultContext(agentName)
 
-	log.Infof("Starting %s\n", agentName)
+	return &ctx
+}
 
-	// Run a periodic timer so we always update StillRunning
-	stillRunning := time.NewTicker(25 * time.Second)
-	agentlog.StillRunning(agentName, warningTime, errorTime)
+func (ctxPtr *baseOsMgrContext) AgentBaseContext() *agentbase.Context {
+	return &ctxPtr.agentBaseContext
+}
 
+func Run(ps *pubsub.PubSub) {
 	// Context to pass around
-	ctx := baseOsMgrContext{
-		globalConfig: types.DefaultConfigItemValueMap(),
-	}
+	ctx := newBaseOsMgrContext()
```

Comment:
All pointers should have the suffix Ptr 
So name it ctxPtr

ctxPtr := newBaseOsMgrContext()